std/random.d

```diff
@@ -2017,7 +2017,7 @@ Returns:
     return a `ref` to the $(D range element), otherwise it will return
     a copy.
  */
-auto ref choice(Range, RandomGen = Random)(auto ref Range range,
+auto ref choice(Range, RandomGen = Random)(inout ref Range range,
```

Comment:
Shouldn't this be `inout auto ref`?

Because I don't think this will compile anymore `choice([1, 2, 3, 4]);`

Reply:
`inout` stands for `auto` in this case. I've added a test for a call by value since your comment.

Reply:
Honestly, this seems like a compiler bug. I've never heard of `inout ref` doing anything similar to `auto ref`, and that behavior does not match what happens with `const` or `immutable`. Also, the way `inout` is designed, it's always supposed to be used on the return value in addition to one of the parameters. So, even if it's true that `inout ref` is supposed to act like `auto ref` with regards to rvalues, it would then be a bug that `inout` isn't used on the return type.

Reply:
The problem is that i use an overload, not the function changed. Actually the fix is wrong...